DEVELOPMENT.md

```diff
@@ -201,6 +201,13 @@ For the Python library interface:
 * Changing the location that a file or directory is created should be considered to be a breaking change. These locations have a tendancy to become
   de-facto parts of the public contract as users build automation that assumes these locations is unchanged.
 
+### Library Dependencies
```

Comment:
Other things that should be considered when adding new dependencies:

### quality of the dependency

- Is the dependency reputable or maintained by a reputable source? Not sure how we can quantify this, but some ideas include:
    -  PyPI download stats?
    - GitHub stars?
    - GitHub dependency graph showing downstream consumers?
- Is it well-maintained? does it release regularly/recently?

### version pinning

How should we pin this new dependency? Does the dependency document their versioning scheme (e.g. semver)?

`deadline-cloud` is both a python library and an application in the case of the CLI. Libraries should strive to accommodate a range of versions to allow consumers to have some flexibility in their closures. Applications should be more opinionated to enforce more reproducible behavior.

In the current structure, we need to balance between both. This is probably a "use your best judgement" decision, but we should prompt the contributor to think about this.

### licensing

*   We also need to ensure the license of the dependency meets our needs
*   We also aim to attribute our dependencies in https://github.com/aws-deadline/deadline-cloud/blob/mainline/THIRD_PARTY_LICENSES

Reply:
Totally agree - these are all good things to consider when selecting 3rd party libs.

Reply:
Added - I reframed the sections The Version pinning one I think needs more work.